### ISSUE TITLE
Segmentation Demo of DIGITS 5.1-dev did not work

### ISSUE THREAD

**Issue by toki2015** [NVIDIA/DIGITS#1277]
I fully followed the operation steps of semantic_segmentation, which was a demonstration stored in examples folder of DIGITS 5.1-dev by using (Caffe ver 0.15.14) based on  CUDA 8.0 (Nvidia Driver 367.57), and  it did not work.  Can anyone tell me why ?

A part of error message : 
```
Creating Layer norm2
I1116 19:37:07.066555 20650 net.cpp:435] norm2 <- pool2
I1116 19:37:07.066566 20650 net.cpp:409] norm2 -> norm2
I1116 19:37:07.066632 20650 net.cpp:144] Setting up norm2
I1116 19:37:07.066648 20650 net.cpp:151] Top shape: 1 256 33 43 (363264)
I1116 19:37:07.066653 20650 net.cpp:159] Memory required for data: 45137776
I1116 19:37:07.066661 20650 layer_factory.hpp:77] Creating layer conv3
I1116 19:37:07.066678 20650 net.cpp:94] Creating Layer conv3
I1116 19:37:07.066685 20650 net.cpp:435] conv3 <- norm2
I1116 19:37:07.066697 20650 net.cpp:409] conv3 -> conv3
F1116 19:37:07.069584 20753 blob.hpp:210] Check failed: data_
*** Check failure stack trace: ***
@     0x7f6cd0d6b5cd  google::LogMessage::Fail()
@     0x7f6cd0d6d433  google::LogMessage::SendToLog()
@     0x7f6cd0d6b15b  google::LogMessage::Flush()
@     0x7f6cd0d6de1e  google::LogMessageFatal::~LogMessageFatal()
I1116 19:37:07.073992 20650 net.cpp:144] Setting up conv3
I1116 19:37:07.074019 20650 net.cpp:151] Top shape: 1 384 33 43 (544896)
I1116 19:37:07.074028 20650 net.cpp:159] Memory required for data: 47317360
I1116 19:37:07.074049 20650 layer_factory.hpp:77] Creating layer relu3
I1116 19:37:07.074064 20650 net.cpp:94] Creating Layer relu3
I1116 19:37:07.074074 20650 net.cpp:435] relu3 <- conv3
I1116 19:37:07.074086 20650 net.cpp:396] relu3 -> conv3 (in-place)
I1116 19:37:07.074103 20650 net.cpp:144] Setting up relu3
I1116 19:37:07.074115 20650 net.cpp:151] Top shape: 1 384 33 43 (544896)
I1116 19:37:07.074121 20650 net.cpp:159] Memory required for data: 49496944
I1116 19:37:07.074129 20650 layer_factory.hpp:77] Creating layer conv4
I1116 19:37:07.074146 20650 net.cpp:94] Creating Layer conv4
I1116 19:37:07.074153 20650 net.cpp:435] conv4 <- conv3
I1116 19:37:07.074167 20650 net.cpp:409] conv4 -> conv4
@     0x7f6cd1437a55  caffe::BasePrefetchingDataLayer<>::InternalThreadEntry()
@     0x7f6cd14d2935  caffe::InternalThread::entry()
@     0x7f6ccf5895d5  (unknown)
@     0x7f6ccf36270a  start_thread
```
Here is the log file of caffe:

[caffe_output_log.txt](https://github.com/NVIDIA/DIGITS/files/594430/caffe_output.log.txt)

**Comment by gheinrich** [comment]
This points towards an issue in data loading. Do your LMDBs look OK? Can you post a screenshot of the dataset job page?

**Comment by l0g1x** [comment]
https://github.com/BVLC/caffe/issues/2251 
https://github.com/BVLC/caffe/tree/master/examples/imagenet

**Comment by kertansul** [comment]
@toki2015 
did you build nvcaffe from source? If yes, you might want to checkout to a specific tag before building.
https://github.com/NVIDIA/DIGITS/issues/1257
https://github.com/kertansul/dl-docker/commit/4d1f2d7b6fb7bb4d15ca7620bd8bec7ca09bd73f

**Comment by toki2015** [comment]
The LMDB looks good (attach screen-shots)

![screenshot from2](https://cloud.githubusercontent.com/assets/11974142/20374470/1f09ebea-acbc-11e6-85a1-6e4fdeb805fe.png)

![screenshot1](https://cloud.githubusercontent.com/assets/11974142/20374448/febd3e78-acbb-11e6-9a35-c9982c65896e.png)

Yes, I build caffe from the  source ( the newest version of NVCaffe)

user@DeepBox:~/caffe$ git status
On branch caffe-0.15
Your branch is up-to-date with 'origin/caffe-0.15'.
nothing to commit, working directory clean
user@DeepBox:~/caffe$ git checkout
Your branch is up-to-date with 'origin/caffe-0.15'.

**Comment by toki2015** [comment]
Sorry to close this issue by a typo.

**Comment by kertansul** [comment]
@toki2015 
I guess there's something wrong with the latest commit on caffe-0.15 that have yet been fixed
`git checkout v0.15.14` and rebuild

**Comment by toki2015** [comment]
You are right. 
This issue was solved by  `git checkout -b v0.15.14 `and rebuild nvcaffe again.
Thank you very much.

**Comment by gheinrich** [comment]
@toki2015 could you help identify which commit on 0.15 branch is causing this regression? Thanks!

**Comment by gheinrich** [comment]
Note: the regression on caffe-0.15 branch was fixed with https://github.com/NVIDIA/caffe/pull/268

**Comment by toki2015** [comment]
Oh... let me take a look about the difference. I am not a developer and wish to find some helpful information ! By the way, nvcaffe still fault to pass through the make runtest based on  CPU (Intel(R) Xeon(R) CPU E5-2695 v2 @ 2.40GHz). I guess it comes from ATLAS (default) package, do you have any idea to solve this issue ?

![screenshot from 2016-11-17 13-14-45](https://cloud.githubusercontent.com/assets/11974142/20421603/47d7a03e-ada8-11e6-9979-1ce101ddf7e5.png)


Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                48
On-line CPU(s) list:   0-47
Thread(s) per core:    2
Core(s) per socket:    12
Socket(s):             2
NUMA node(s):          2
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 62
Model name:            Intel(R) Xeon(R) CPU E5-2695 v2 @ 2.40GHz
Stepping:              4
CPU MHz:               2393.906
CPU max MHz:           3200.0000
CPU min MHz:           1200.0000
BogoMIPS:              4801.45
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              30720K
NUMA node0 CPU(s):     0-11,24-35
NUMA node1 CPU(s):     12-23,36-47
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm epb tpr_shadow vnmi flexpriority ept vpid fsgsbase smep erms xsaveopt dtherm ida arat pln pts

**Comment by gheinrich** [comment]
If you don't mind I will close the issue since it is now resolved on caffe 0.15 branch.
